ever_given/ever_given/wrapper.py

```diff
@@ -2,53 +2,57 @@
 import shlex
 import sys
 from pathlib import Path
+import typing
 
-import docker
+from .engines import GUEST_OUTPUT_DIR, REGISTERED_ENGINES
 
 from . import log_processing
+from .utils import LogHandlerBase
 
 GUEST_INPUT_DIR = Path("/mnt") / "inputs"
-GUEST_OUTPUT_DIR = Path("/mnt") / "outputs"
 
 
-def _guest_input_path(filename):
+def _guest_input_path(filename: str) -> Path:
     return GUEST_INPUT_DIR / Path(filename).name
 
 
-def prepare_commandline(command, args_dict):
-    return (
-        command
-        + " "
-        + " ".join(
-            [f"--{key} {shlex.quote(str(value))}" for key, value in args_dict.items()]
-        )
-    )
+def prepare_command_list(command: str, args_dict: typing.Dict[str, str]):
+    return [command] + [
+        f"--{key} {shlex.quote(str(value))}" for key, value in args_dict.items()
```

Comment:
I'm not sure this will work - if we're passing the command list as a list, we might need to split the args so we can bypass shell parsing.